getSimulationSCSH.m

```diff
@@ -65,10 +67,15 @@
                     ny = size(SP.dCydxi,2);
                     dtmpdxi = arrayfun(@(x,y) diag(squeeze(SP.dCydxi(x,:,:,y))),repmat(1:nt,[np,1]),...
                         repmat(transpose(1:np),[1,nt]),'UniformOutput',false);
-                    dmydxi = bsxfun(@times,my,SP.dmydxi) ...
-                        + bsxfun(@times,my,permute(reshape([dtmpdxi{:}]/2,...
-                        [ny,np,nt]),[3,1,2]));
+                    dtmpdxi = permute(reshape([dtmpdxi{:}]/2, [ny,np,nt]),[3,1,2]);
+                    dmydxi = [dmydxi; bsxfun(@times,my_new,SP.dmydxi) ...
+                        + bsxfun(@times,my_new, dtmpdxi)];
                     dmydxi(isnan(dmydxi)) = 0;
+                    dtmpCydxi = bsxfun(@plus, repmat(permute(dtmpdxi, [1,2,4,3]),...
+                        1, 1, size(SP.Cy,3), 1), repmat(permute(dtmpdxi, [1,4,2,3]),...
+                        1, size(SP.Cy,3), 1, 1));
+                    dCydxi = [dCydxi; bsxfun(@times, exp(tmpCy) .* (exp(SP.Cy) - ones(size(SP.Cy))), dtmpCydxi)...
+                        + bsxfun(@times, exp(tmpCy), bsxfun(@times, exp(SP.Cy), SP.dCydxi))];
```

Comment:
Seems reasonable as well...

Reply:
At a first glance, this seems to be okay for full covariance matrices as well...